src/methods.jl

```diff
@@ -120,3 +120,54 @@ julia> flat_correct(frame, flat)
 * [`flat_correct!`](@ref)
 """
 flat_correct(frame::AbstractArray, flat_frame::AbstractArray; kwargs...) = flat_correct!(deepcopy(frame), flat_frame; kwargs...)
+
+
+"""
+    trim(frame::AbstractArray, idxs)
+
+Trims the frame to remove the region specified by idxs.
+
+This function trims the array in a manner such that final array should be rectangular.
+The indices follow standard Julia convention, so `(:, 45:60)` trims all columns from 45 to 60 and `(1:20, :)` trims all the rows from 1 to 20.
+
+# Examples
+```jldoctest
+julia> frame = ones(5, 5);
+
+julia> trim(frame, (:, 2:5))
+5×1 Array{Float64,2}:
+ 1.0
+ 1.0
+ 1.0
+ 1.0
+ 1.0
+
+```
+
+# See Also
+[`trim!`](@ref)
+"""
+trim(frame::AbstractArray, idxs) = collect(trim!(frame, idxs))
```

Comment:
`copy` or `deepcopy` are probably more idiomatic than `collect`?

Reply:
So, should we switch to something like 
```
trim(...) = deepcopy(trim!(...))
```
Right?

Reply:
I tried both these versions, here are the results (I defined new function `trim1` with `deepcopy`)
```
trim1(frame::AbstractArray, idxs) = deepcopy(trim!(frame, idxs))
```

The results were:
```
julia> x = rand(4096, 4096);

julia> @benchmark trim(x, (:, 3000:4096))
BenchmarkTools.Trial: 
  memory estimate:  94.50 MiB
  allocs estimate:  22993
  --------------
  minimum time:     77.609 ms (0.00% GC)
  median time:      78.180 ms (0.39% GC)
  mean time:        83.516 ms (6.98% GC)
  maximum time:     147.196 ms (47.21% GC)
  --------------
  samples:          60
  evals/sample:     1

julia> @benchmark trim1(x, (:, 3000:4096))
BenchmarkTools.Trial: 
  memory estimate:  128.81 MiB
  allocs estimate:  23002
  --------------
  minimum time:     92.075 ms (0.00% GC)
  median time:      103.485 ms (0.31% GC)
  mean time:        102.107 ms (7.45% GC)
  maximum time:     166.797 ms (44.01% GC)
  --------------
  samples:          50
  evals/sample:     1
```
Memory and time utilised in `trim1` with `deepcopy` is more as compared to `trim`, so I guess it's better to use `trim`. What causes this kind of difference in utilised memory and time?

Reply:
Did you try `copy`?

Reply:
`copy` is faster!
```
julia> trim2(frame::AbstractArray, idxs) = copy(trim!(frame, idxs));

julia> x = randn(4096, 4096);

julia> @benchmark trim2(x, (:, 3000:4096))
BenchmarkTools.Trial: 
  memory estimate:  94.50 MiB
  allocs estimate:  22995
  --------------
  minimum time:     65.307 ms (0.00% GC)
  median time:      67.059 ms (0.40% GC)
  mean time:        71.168 ms (7.81% GC)
  maximum time:     136.852 ms (51.54% GC)
  --------------
  samples:          71
  evals/sample:     1
```

Reply:
I missed interpolating the variables, let me do it with interpolation!

Reply:
`trim` - with collect
`trim1` - with deepcopy
`trim2` - with copy
```
julia> x = rand(4096, 4096);

julia> @benchmark trim($x, $((:, 3000:4096)))
BenchmarkTools.Trial: 
  memory estimate:  94.50 MiB
  allocs estimate:  22993
  --------------
  minimum time:     43.108 ms (0.00% GC)
  median time:      44.161 ms (0.59% GC)
  mean time:        46.359 ms (6.48% GC)
  maximum time:     91.735 ms (52.76% GC)
  --------------
  samples:          108
  evals/sample:     1

julia> @benchmark trim1($x, $((:, 3000:4096)))
BenchmarkTools.Trial: 
  memory estimate:  128.81 MiB
  allocs estimate:  23002
  --------------
  minimum time:     52.172 ms (0.00% GC)
  median time:      53.200 ms (0.36% GC)
  mean time:        56.204 ms (6.78% GC)
  maximum time:     102.094 ms (48.88% GC)
  --------------
  samples:          89
  evals/sample:     1

julia> @benchmark trim2($x, $((:, 3000:4096)))
BenchmarkTools.Trial: 
  memory estimate:  94.50 MiB
  allocs estimate:  22995
  --------------
  minimum time:     36.425 ms (0.00% GC)
  median time:      37.791 ms (0.52% GC)
  mean time:        40.183 ms (7.50% GC)
  maximum time:     85.394 ms (57.44% GC)
  --------------
  samples:          125
  evals/sample:     1
```

Reply:
Used `copy` as it was the fastest amongst all!